### PR TITLE
ESB samples nRF52833 DK support

### PR DESCRIPTION
**Pull request by leewkb4567**
Add nRF52833 DK support to ESB samples.

JIRA: NCSIDB-640

### CODE REVIEW

**Review by umapraseeda — APPROVED** [review]

**Comment by grochu** [comment]
Will the code section in https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/esb/esb.c#L375 be executed for nRF52833?

**Review by grochu — COMMENTED** [review]
Please check the _nrf/subsys/esb/esb.c_ file for any additional code that applies for nRF52833 and may be compiled out.

**Comment by leewkb4567** [comment]
> Please check the _nrf/subsys/esb/esb.c_ file for any additional code that applies for nRF52833 and may be compiled out.

The ESB code looks OK. The ESB_BITRATE_2MBPS_BLE part is not very clean but it works too.

**Comment by grochu** [comment]
> > Will the code section in https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/esb/esb.c#L375 be executed for nRF52833?
> 
> I suppose so. Setting to bitrate ESB_BITRATE_1MBPS_BLE, the ptx and prx samples work fine.

I was thinking about the `ESB_BITRATE_250KBPS` specifically. The esb.c line 369 code checks `CONFIG_SOC_SERIES_NRF51X` symbol, which will not be defined for nRF52833 so the switch option will not be compiled - although I am not 100% sure it should. Should this bitrate supported on the nRF52 devices? Maybe this has been wrong already for nRF52832/840?

**Comment by grochu** [comment]
@leewkb4567 it should be OK to merge from the development point of view. My concern is the test coverage. Shall we expand the test setup with a nRf52833 DK? @shantha-14 could you please comment?

**Comment by tejlmand** [comment]
@leewkb4567 seems this should have an entry in the changelog here: https://github.com/nrfconnect/sdk-nrf/blob/main/doc/nrf/releases/release-notes-changelog.rst

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -191,6 +191,13 @@ Libraries for networking
 
     * An issue where downloads of COAP URIs would fail when they contained multiple path elements.
 
+ESB
+---
+
+* Updated:
+
+    * Fixed a compilation error for nRF52833.
+
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:200]:
```suggestion
Other libraries
---------------

* Updated:

  * :ref:`esb_readme`:

    * Fixed a compilation error for nRF52833.

```

```

**Review by umapraseeda — CHANGES_REQUESTED** [review]

**Comment by leewkb4567** [comment]
Rebased to current main for changelog changes.

@umapraseeda Your comment is addressed.

**Review by umapraseeda — APPROVED** [review]
LGTM. Thanks!

**Review by tejlmand — APPROVED** [review]

**Pull request merged**